File: Kotlin/app/src/main/java/com/example/tictactoe/StartScreen.kt
package com.example.tictactoe

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.compose.ui.Alignment
import androidx.compose.foundation.clickable
import androidx.compose.foundation.border
import com.example.tictactoe.ui.theme.*

data class GameSettings(
    val mode: GameMode,
    val boardSize: Int = 3,
    val player1Color: Color,
    val player2Color: Color,
    val player1Shape: PlayerShape,
    val player2Shape: PlayerShape
)

enum class GameMode {
    SINGLE_MODE,
    DOUBLE_MODE
}

enum class PlayerShape(val symbol: String) {
    CROSS("X"), CIRCLE("O"), SQUARE("■"), TRIANGLE("▲"), STAR("★")
}

@Composable
fun StartScreen(onStartGame: (GameSettings) -> Unit) {

    Box(modifier = Modifier.fillMaxSize().background(MaterialTheme.colorScheme.background)) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(16.dp),
            verticalArrangement = Arrangement.Center,
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            // Tytuł
            Text(
                text = "Select game mode and settings",
                style = MaterialTheme.typography.headlineLarge,
                color = MaterialTheme.colorScheme.onBackground,
                modifier = Modifier.padding(bottom = 32.dp)
            )

            var selectedMode by remember { mutableStateOf(GameMode.SINGLE_MODE) }
            var selectedBoardSize by remember { mutableStateOf(3) }
            var player1Color by remember { mutableStateOf(Yellow200) }
            var player2Color by remember { mutableStateOf(Blue200) }
            var player1Shape by remember { mutableStateOf(PlayerShape.CROSS) }
            var player2Shape by remember { mutableStateOf(PlayerShape.CIRCLE) }

            // Tryb gry
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceEvenly
            ) {

                Button(
                    onClick = { selectedMode = GameMode.SINGLE_MODE },
                    colors = ButtonDefaults.buttonColors(
                        containerColor = if (selectedMode == GameMode.SINGLE_MODE) Gray else LightGray
                    )
                ) {
                    Text("Single Mode")
                }

                Button(
                    onClick = { selectedMode = GameMode.DOUBLE_MODE },
                    colors = ButtonDefaults.buttonColors(
                        containerColor = if (selectedMode == GameMode.DOUBLE_MODE) Gray else LightGray
                    )
                ) {
                    Text("Double Mode")
                }
            }

            Spacer(modifier = Modifier.height(32.dp))

            // Wybór rozmiaru planszy
            Text(text = "Choose Board Size", color = MaterialTheme.colorScheme.onBackground)
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceEvenly
            ) {
                listOf(3, 4, 5).forEach { size ->
                    Button(
                        onClick = { selectedBoardSize = size },
                        colors = ButtonDefaults.buttonColors(
                            containerColor = if (selectedBoardSize == size) Gray else LightGray
                        )
                    ) {
                        Text("${size}x$size")
                    }
                }
            }

            Spacer(modifier = Modifier.height(32.dp))

            if (selectedMode == GameMode.SINGLE_MODE) {
                // Opcje dla trybu Single Mode
                Text(text = "Choose Color and Shape", color = MaterialTheme.colorScheme.onBackground)

                // Wybór koloru dla gracza
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceEvenly
                ) {
                    listOf(Yellow200, Red200, Blue200, Green200, Purple200).forEach { color ->
                        Box(
                            modifier = Modifier
                                .size(50.dp)
                                .background(color)
                                .clickable { player1Color = color }
                                .border(2.dp, if (player1Color == color) MaterialTheme.colorScheme.onBackground else MaterialTheme.colorScheme.background)
                        )
                    }
                }

                Spacer(modifier = Modifier.height(16.dp))

                // Wybór kształtu dla gracza
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceEvenly
                ) {
                    PlayerShape.values().forEach { shape ->
                        Button(
                            onClick = { player1Shape = shape },
                            colors = ButtonDefaults.buttonColors(
                                containerColor = if (player1Shape == shape) Gray else LightGray
                            )
                        ) {
                            Text(shape.symbol)
                        }
                    }
                }

                // Ustal losowy kolor i kształt dla komputera (inne niż gracza)
                LaunchedEffect(player1Color, player1Shape) {
                    val colors = listOf(Yellow200, Red200, Blue200, Green200, Purple200) - player1Color
                    val shapes = PlayerShape.values().toList() - player1Shape
                    player2Color = colors.random()
                    player2Shape = shapes.random()
                }

            } else {
                // Opcje dla trybu Double Mode
                Text(text = "Player 1 - Choose Color and Shape", color = MaterialTheme.colorScheme.onBackground)

                // Wybór koloru i kształtu dla Player 1
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceEvenly
                ) {
                    listOf(Yellow200, Red200, Blue200, Green200, Purple200).forEach { color ->
                        Box(
                            modifier = Modifier
                                .size(50.dp)
                                .background(color)
                                .clickable {
                                    if (color != player2Color) player1Color = color
                                }
                                .border(2.dp, if (player1Color == color) MaterialTheme.colorScheme.onBackground else MaterialTheme.colorScheme.background)
                        )
                    }
                }

                Spacer(modifier = Modifier.height(16.dp))

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceEvenly
                ) {
                    PlayerShape.values().forEach { shape ->
                        Button(
                            onClick = {
                                if (shape != player2Shape) player1Shape = shape
                            },
                            colors = ButtonDefaults.buttonColors(
                                containerColor = if (player1Shape == shape) Gray else LightGray
                            )
                        ) {
                            Text(shape.symbol)
                        }
                    }
                }

                Spacer(modifier = Modifier.height(32.dp))
                Text(text = "Player 2 - Choose Color and Shape", color = MaterialTheme.colorScheme.onBackground)

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceEvenly
                ) {
                    listOf(Yellow200, Red200, Blue200, Green200, Purple200).forEach { color ->
                        Box(
                            modifier = Modifier
                                .size(50.dp)
                                .background(color)
                                .clickable {
                                    if (color != player1Color) player2Color = color
                                }
                                .border(2.dp, if (player2Color == color) MaterialTheme.colorScheme.onBackground else MaterialTheme.colorScheme.background)
                        )
                    }
                }

                Spacer(modifier = Modifier.height(16.dp))

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceEvenly
                ) {
                    PlayerShape.values().forEach { shape ->
                        Button(
                            onClick = {
                                if (shape != player1Shape) player2Shape = shape
                            },
                            colors = ButtonDefaults.buttonColors(
                                containerColor = if (player2Shape == shape) Gray else LightGray
                            )
                        ) {
                            Text(shape.symbol)
                        }
                    }
                }
            }

            Spacer(modifier = Modifier.height(64.dp))

            // Przycisk PLAY
            Button(
                onClick = {
                    onStartGame(GameSettings(
                        mode = selectedMode,
                        boardSize = selectedBoardSize,
                        player1Color = player1Color,
                        player2Color = player2Color,
                        player1Shape = player1Shape,
                        player2Shape = player2Shape
                    ))
                },
                modifier = Modifier
                    .align(Alignment.CenterHorizontally)
                    .size(100.dp, 50.dp),
                colors = ButtonDefaults.buttonColors(
                    contentColor = Color.Black)
            ) {
                Text("PLAY")
            }
        }
    }
}

File: flutter_tictactoe/lib/StartScreen.dart
import 'package:flutter/material.dart';
import 'ui/theme/colors.dart';
import 'ui/theme/typography.dart';
import 'GameScreen.dart';
import 'GameSettings.dart';

class StartScreen extends StatefulWidget {
  @override
  _StartScreenState createState() => _StartScreenState();
}

class _StartScreenState extends State<StartScreen> {
  GameMode selectedMode = GameMode.SINGLE_MODE;
  int selectedBoardSize = 3;
  Color player1Color = AppColors.yellow200;
  Color player2Color = AppColors.blue200;
  PlayerShape player1Shape = PlayerShape.CROSS;
  PlayerShape player2Shape = PlayerShape.CIRCLE;

  void startGame() {
    // Losowe ustawienia dla komputera w trybie Single Mode
    if (selectedMode == GameMode.SINGLE_MODE) {
      player2Color = Colors.primaries.where((color) => color != player1Color).first;
      player2Shape = PlayerShape.values.where((shape) => shape != player1Shape).first;
    }

    Navigator.push(
      context,
      MaterialPageRoute(
        builder: (context) => GameScreen(
          gameSettings: GameSettings(
            mode: selectedMode,
            boardSize: selectedBoardSize,
            player1Color: player1Color,
            player2Color: player2Color,
            player1Shape: player1Shape,
            player2Shape: player2Shape,
          ),
        ),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Theme.of(context).colorScheme.background,
      body: Center(
        child: Padding(
          padding: const EdgeInsets.all(16.0),
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              Text(
                "Select game mode and settings",
                style: AppTypography.headlineLarge.copyWith(color: Theme.of(context).colorScheme.onBackground),
                textAlign: TextAlign.left,
              ),
              const SizedBox(height: 32),

              // Tryb gry
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: [
                  ElevatedButton(
                    onPressed: () => setState(() => selectedMode = GameMode.SINGLE_MODE),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: selectedMode == GameMode.SINGLE_MODE ? AppColors.gray : AppColors.lightGray,
                      foregroundColor: Theme.of(context).colorScheme.onPrimary,
                    ),
                    child: Text("Single Mode"),
                  ),
                  ElevatedButton(
                    onPressed: () => setState(() => selectedMode = GameMode.DOUBLE_MODE),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: selectedMode == GameMode.DOUBLE_MODE ? AppColors.gray : AppColors.lightGray,
                      foregroundColor: Theme.of(context).colorScheme.onPrimary,
                    ),
                    child: Text("Double Mode"),
                  ),
                ],
              ),

              const SizedBox(height: 32),

              // Rozmiar planszy
              Text("Choose Board Size", style: AppTypography.bodyMedium.copyWith(color: Theme.of(context).colorScheme.onBackground),),
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: [3, 4, 5].map((size) {
                  return ElevatedButton(
                    onPressed: () => setState(() => selectedBoardSize = size),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: selectedBoardSize == size ? AppColors.gray : AppColors.lightGray,
                      foregroundColor: Theme.of(context).colorScheme.onPrimary,
                    ),
                    child: Text("${size}x$size"),
                  );
                }).toList(),
              ),

              const SizedBox(height: 32),

              // Kolor i kształt dla trybu Single Mode
              if (selectedMode == GameMode.SINGLE_MODE) ...[
                Text("Choose Color and Shape", style: AppTypography.bodyMedium.copyWith(color: Theme.of(context).colorScheme.onBackground)),

                // Wybór koloru
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: [AppColors.yellow200, AppColors.red200, AppColors.blue200, AppColors.green200, AppColors.purple200].map((color) {
                    return GestureDetector(
                      onTap: () => setState(() => player1Color = color),
                      child: Container(
                        width: 50,
                        height: 50,
                        decoration: BoxDecoration(
                          color: color,
                          border: Border.all(
                            color: player1Color == color ? Theme.of(context).colorScheme.onBackground : Theme.of(context).colorScheme.background,
                            width: 2,
                          ),
                        ),
                      ),
                    );
                  }).toList(),
                ),

                const SizedBox(height: 16),

                // Wybór kształtu
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: PlayerShape.values.map((shape) {
                    return ElevatedButton(
                      onPressed: () => setState(() => player1Shape = shape),
                      style: ElevatedButton.styleFrom(
                        backgroundColor: player1Shape == shape ? AppColors.gray : AppColors.lightGray,
                        foregroundColor: Theme.of(context).colorScheme.onPrimary,
                      ),
                      child: Text(shape.symbol),
                    );
                  }).toList(),
                ),
              ],

              if (selectedMode == GameMode.DOUBLE_MODE) ...[
                // Ustawienia dla gracza 1
                Text("Player 1 - Choose Color and Shape", style: AppTypography.bodyMedium.copyWith(color: Theme.of(context).colorScheme.onBackground)),
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: [AppColors.yellow200, AppColors.red200, AppColors.blue200, AppColors.green200, AppColors.purple200].map((color) {
                    return GestureDetector(
                      onTap: () => setState(() {
                        if (color != player2Color) player1Color = color;
                      }),
                      child: Container(
                        width: 50,
                        height: 50,
                        decoration: BoxDecoration(
                          color: color,
                          border: Border.all(
                            color: player1Color == color ? Theme.of(context).colorScheme.onBackground : Theme.of(context).colorScheme.background,
                            width: 2,
                          ),
                        ),
                      ),
                    );
                  }).toList(),
                ),

                const SizedBox(height: 16),

                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: PlayerShape.values.map((shape) {
                    return ElevatedButton(
                      onPressed: () => setState(() {
                        if (shape != player2Shape) player1Shape = shape;
                      }),
                      style: ElevatedButton.styleFrom(
                        backgroundColor: player1Shape == shape ? AppColors.gray : AppColors.lightGray,
                        foregroundColor: Theme.of(context).colorScheme.onPrimary,
                      ),
                      child: Text(shape.symbol),
                    );
                  }).toList(),
                ),

                const SizedBox(height: 32),

                Text("Player 2 - Choose Color and Shape", style: AppTypography.bodyMedium.copyWith(color: Theme.of(context).colorScheme.onBackground)),
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: [AppColors.yellow200, AppColors.red200, AppColors.blue200, AppColors.green200, AppColors.purple200].map((color) {
                    return GestureDetector(
                      onTap: () => setState(() {
                        if (color != player1Color) player2Color = color;
                      }),
                      child: Container(
                        width: 50,
                        height: 50,
                        decoration: BoxDecoration(
                          color: color,
                          border: Border.all(
                            color: player2Color == color ? Theme.of(context).colorScheme.onBackground : Theme.of(context).colorScheme.background,
                            width: 2,
                          ),
                        ),
                      ),
                    );
                  }).toList(),
                ),

                const SizedBox(height: 16),

                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: PlayerShape.values.map((shape) {
                    return ElevatedButton(
                      onPressed: () => setState(() {
                        if (shape != player1Shape) player2Shape = shape;
                      }),
                      style: ElevatedButton.styleFrom(
                        backgroundColor: player2Shape == shape ? AppColors.gray : AppColors.lightGray,
                        foregroundColor: Theme.of(context).colorScheme.onPrimary,
                      ),
                      child: Text(shape.symbol),
                    );
                  }).toList(),
                ),
              ],

              const SizedBox(height: 32),

              // Przycisk PLAY
              ElevatedButton(
                onPressed: startGame,
                style: ElevatedButton.styleFrom(
                  minimumSize: Size(100, 50),
                  backgroundColor: Theme.of(context).colorScheme.primary,
                  foregroundColor: Theme.of(context).colorScheme.onPrimary,
                ),
                child: Text("PLAY"),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
